src/Core/Analysis/IAnalysisIssue.cs

```diff
@@ -100,4 +100,12 @@ public enum AnalysisIssueType
         Bug,
         Vulnerability
     }
+
+    public static class IAnalysisIssueExtensions
+    {
+        public static bool IsFileLevel(this IAnalysisIssue issue)
```

Comment:
Shouldn't this use `IAnalysisIssueBase` since that is the interface that has `PrimaryLocation`?